internal/sys/ptr.go

```diff
@@ -1,43 +1,57 @@
 package sys
 
 import (
+	"math"
 	"unsafe"
 
 	"github.com/cilium/ebpf/internal/unix"
 )
 
-// NewPointer creates a 64-bit pointer from an unsafe Pointer.
-func NewPointer(ptr unsafe.Pointer) Pointer {
+// UnsafePointer creates a 64-bit pointer from an unsafe Pointer.
```

Comment:
This makes some cases stutter (`sys.UnsafePointer(unsafe.Pointer(&info))`) and makes for longer lines overall. It still allocates, just not necessarily on the heap. :wink: I think NewPointer was just fine, it still returns 'a new Pointer'.

Reply:
True about the stuttering! Pointer is a value type, so in my mind that means it doesn't allocate on it's own. Contrast that with NewSlicePointer, etc.